### PR TITLE
update readme with links and an image

### PR DESCRIPTION
**Pull request by clbarrell**
Adds more content to the readme describing templates, the gallery and what they are.

Can someone that has the screenshot tool maybe make the screenshot nice with the padding and background? Not 100% necessary I guess.

Open to copy suggestions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,22 @@
-# Templates
+# Buildkite Pipeline Templates
```
Comment (olyism) [inline README.md:1]: I think we're using the term "templates" with lower case.

```suggestion
# Buildkite Pipeline templates
```

**Comment by olyism** [comment]
How's this screenshot?

screenshot

File: template.png
Comment (olyism) [inline template.png:1]: > Can someone that has the screenshot tool maybe make the screenshot nice with the padding and background?

Sure! How about this screenshot?

screen

Also might be good to show a dev tool pipeline over Swift, so people won't misunderstood us as mobile only?

Comment (clbarrell) [reply template.png:1]: Good idea, thanks!

File: README.md [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+
+Each template is a Buildkite pipeline performing a specific task using typical technologies and programming languages. For example, the [Ruby CI template](https://buildkite.com/pipelines/templates/ci/ruby-ci) is an example of doing continuous integration (CI) in Ruby using Rubocop and Rspec.
+
+## How do they work?
+
+Each template is stored as a root level folder, containing a pipeline YAML file and readme. When viewed on the Buildkite website, users can click 'Use Template' to create a new pipeline in their account using the templated pipeline steps.
+
+## Requesting templates
+
+Request a new template by [opening an issue](https://github.com/buildkite/templates/issues/new?template=pipeline-template-request.md). Please let us know what types of use cases and technologies you wnat to see covered.
```
Comment (dannymidnight) [inline README.md:18]:
```suggestion
Request a new template by [opening an issue](https://github.com/buildkite/templates/issues/new?template=pipeline-template-request.md). Please let us know what types of use cases and technologies you want to see covered.
```

File: README.md [hunk 3]
```diff
@@ -1,7 +1,22 @@
-# Templates
+# Buildkite Pipeline templates
```
Comment (dannymidnight) [inline README.md:1]: Should this be lowercased? 

```suggestion
# Buildkite pipeline templates
```

**Review by dannymidnight — APPROVED** [review]
Looks good 👍

**Review by olyism — APPROVED** [review]
LGTM ✅

**Pull request merged**